src/net.cpp

```diff
@@ -1734,7 +1734,15 @@ void ThreadOpenConnections()
                 LogPrint("net", "Making feeler connection to %s\n", addrConnect.ToString());
             }
 
-            OpenNetworkConnection(addrConnect, (int)setConnected.size() >= std::min(nMaxConnections - 1, 2), &grant, NULL, false, fFeeler);
+            int64_t nStartTime = GetTimeMicros();
+
+            bool fRet = OpenNetworkConnection(addrConnect, (int)setConnected.size() >= std::min(nMaxConnections - 1, 2), &grant, NULL, false, fFeeler);
+
+            // Retry feeler connection if OpenNetworkConnection fails in under 1ms.
+            // Avoids issues with feelers to IP space we don't know how to route.
+            if (!fRet && GetTimeMicros() - nStartTime < 1000) {
+                nNextFeeler = GetTime();
```

Comment:
nNextFeeler is in microseconds, so you should reuse the value of the call to GetTimeMicros instead (or just assign 0, which will have the same effect).